src/main/java/bio/terra/service/iam/sam/SamRetry.java

```diff
@@ -33,40 +33,77 @@
         this.operationTimeout = now().plusSeconds(operationTimeoutSeconds);
     }
 
-    <T> T perform(SamFunction<T> function) throws InterruptedException {
+    static <T> T retry(ConfigurationService configService, SamFunction<T> function)
+        throws InterruptedException {
+        SamRetry samRetry = new SamRetry(configService);
+        return samRetry.perform(function);
+    }
+
+    static void retryVoid(ConfigurationService configService, SamVoidFunction function)
```

Comment:
Since java has overloading, this could also be called `retry()`.

Reply:
Much cleaner, thank you!